### PR TITLE
Update MockObject class in Magento Framework

### PR DESCRIPTION
**Pull request by hws47a**
### Description (*)
Replace outdated PSR-0 `\PHPUnit_Framework_MockObject_MockObject` with current PSR-2 class name `\PHPUnit\Framework\MockObject\MockObject`

### Related Pull Requests
#26775 

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by lbajsarowicz — DISMISSED** [review]
Could you remove spaces around pipe ` | `?

File: lib/internal/Magento/Framework/App/Test/Unit/Config/MetadataConfigTypeProcessorTest.php [hunk 1]
```diff
@@ -11,7 +11,7 @@
 use Magento\Framework\App\Config\Data\ProcessorInterface;
 use Magento\Framework\App\Config\Initial;
 use Magento\Framework\App\Config\MetadataConfigTypeProcessor;
-use \PHPUnit_Framework_MockObject_MockObject as MockObject;
+use \PHPUnit\Framework\MockObject\MockObject as MockObject;
```
Comment (Nazar65) [inline lib/internal/Magento/Framework/App/Test/Unit/Config/MetadataConfigTypeProcessorTest.php]:
```suggestion
use \PHPUnit\Framework\MockObject\MockObject;
```
Could you please also remove `as `where this is not necessary ?

**Review by Nazar65 — CHANGES_REQUESTED** [review]

**Comment by hws47a** [comment]
@lbajsarowicz it was already removed. I took the code from the previous PR

**Comment by hws47a** [comment]
@Nazar65 done

**Review by Nazar65 — APPROVED** [review]

**Comment by sivaschenko** [comment]
@magento run Static Tests, Unit Tests

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Hi @hws47a thanks for the pull request! There are multiple static and unit test failures, can you please fix those

**Comment by sivaschenko** [comment]
@hws47a the pull request intention is great, however, I think it may be hard to fix all the static test for all the 632 files, also it's not comfortable to review and potentially a lot of conflicts can occur during the processing.

I'd recommend dividing this contribution into several PRs ~50 files each. That will make these contributions much easier to process

**Comment by ihor-sviziev** [comment]
Hi @hws47a,
I'm closing this PR as it contains too many changed files and already have conficts.
Please create separate PRs per module, so it will be much easier to fix the issues and deliver it fast.

Thank you for your contritbution!

**Pull request closed without merging**